developer_manual/digging_deeper/groupware/mail_provider.rst

```diff
@@ -0,0 +1,242 @@
+
+.. _mail-providers:
+
+========================================
+Mail Provider Interface
+========================================
+
+Nextcloud apps can use, and register the mail provider interface, to provide or consume mail service functionality to and from other apps. 
+
+Access to mail services from another app is provided by the Mail Manager class. Using this class your app can find available mail providers and services. Please see the appropriate section below for more details.
+
+The reverse is also possible, one can register an app to supply mail services to other apps. This is accomplished by implementing a custom Mail Provider class and Mail Service class then registering your provider with the system. Please see the appropriate section below for more details.
+
+.. _mail-provider-terminology:
+
+Terminology
+-----------
+
+For clarificaftion this is a reference of terminology used.
+
+1. *Provider* - this is the app that provides mail services e.g. "BigHostingApp", this can a general app for a specific protocol
```

Comment:
```suggestion
1. *Provider* - this is the app that provides mail services e.g. "BigHostingApp", this can be an app for a specific protocol
```

Reply:
No worries, I know my spelling is horrible